alot/crypto.py

```diff
@@ -81,53 +54,64 @@ def get_key(keyid, validate=False, encrypt=False, sign=False,
     :param signed_only: only return keys  whose uid is signed (trusted to
         belong to the key)
     :type signed_only: bool
-    :rtype: gpgme.Key
+    :rtype: gpg key object
     """
-    ctx = gpgme.Context()
+    ctx = gpg.core.Context()
     try:
         key = ctx.get_key(keyid)
         if validate:
             validate_key(key, encrypt=encrypt, sign=sign)
-    except gpgme.GpgmeError as e:
-        if e.code == gpgme.ERR_AMBIGUOUS_NAME:
+    except gpg.errors.GPGMEError as e:
+        if e.getcode() == gpg.errors.AMBIGUOUS_NAME:
             # When we get here it means there were multiple keys returned by
             # gpg for given keyid. Unfortunately gpgme returns invalid and
             # expired keys together with valid keys. If only one key is valid
             # for given operation maybe we can still return it instead of
             # raising exception
+            #
             keys = list_keys(hint=keyid)
+
             valid_key = None
-            for k in keys:
-                try:
-                    validate_key(k, encrypt=encrypt, sign=sign)
-                except GPGProblem:
-                    # if the key is invalid for given action skip it
-                    continue
-
-                if valid_key:
-                    # we have already found one valid key and now we find
-                    # another? We really received an ambiguous keyid
+            try:
+                for k in keys:
+                    try:
+                        validate_key(k, encrypt=encrypt, sign=sign)
+                    except GPGProblem:
+                        # if the key is invalid for given action skip it
+                        continue
+
+                    if valid_key:
+                        # we have already found one valid key and now we find
+                        # another? We really received an ambiguous keyid
+                        raise GPGProblem(
+                            "More than one key found matching this filter. "
+                            "Please be more specific "
+                            "(use a key ID like 4AC8EE1D).",
+                            code=GPGCode.AMBIGUOUS_NAME)
+                    valid_key = k
+            except gpg.errors.GPGMEError as e:
```

Comment:
no. I don't know what i was thinking, not `validate_key`, in the generator returned by `list_key`